### PR TITLE
Threadlocal logging - some improvements

### PR DESCRIPTION
**Pull request by seime**
Based #74 

Commands/updates sent from a rule now logs with the originating rule name/log name like

`[My rule] SendCommand MyLightBUlbItem 100`

Also works for embedded (and nested) timers originating from rules:

`[Rule log name/TimerName/NestedTimer] SendCommand targetItem nestedTimedCommand`

See test `JRuleTimerTest`

### CODE REVIEW

File: src/main/java/org/openhab/automation/jrule/internal/JRuleUtil.java [hunk 1]
```diff
@@ -90,13 +92,19 @@ public static <T> CompletableFuture<T> delayedExecution(long delay, TimeUnit uni
...
-            TimeUnit unit) {
+            TimeUnit unit, JRuleExecutionContext context) {
         CompletableFuture<T> completableFuture = new CompletableFuture<>();
         scheduler.schedule((() -> {
-            command.get().thenAccept(completableFuture::complete).exceptionally(t -> {
-                completableFuture.completeExceptionally(t);
-                return null;
-            });
+            try {
+                JRule.JRULE_EXECUTION_CONTEXT.set(context);
```
Comment (querdenker2k) [inline src/main/java/org/openhab/automation/jrule/internal/JRuleUtil.java:99]: its hard to get a full overview inside the webui, but isn't it possible without using this static field var?

File: src/main/java/org/openhab/automation/jrule/internal/handler/JRuleEventHandler.java [hunk 2]
```diff
@@ -177,7 +179,7 @@ public void sendCommand(String itemName, Command command) {
         if (eventPublisher == null) {
             return;
         }
-        logInfo("SendCommand itemName: {} command: {}", itemName, command);
+        logInfo("SendCommand {} {}", itemName, command);
```
Comment (querdenker2k) [inline src/main/java/org/openhab/automation/jrule/internal/handler/JRuleEventHandler.java:182]: some more separation like "sendCommand '{}' on '{}'"

File: src/main/java/org/openhab/automation/jrule/internal/handler/JRuleEventHandler.java [hunk 3]
```diff
@@ -248,7 +250,7 @@ public void postUpdate(String itemName, State state) {
         if (eventPublisher == null) {
             return;
         }
-        logInfo("PostUpdate itemName: {} state: {}", itemName, state);
+        logInfo("PostUpdate {} {}", itemName, state);
```
Comment (querdenker2k) [inline src/main/java/org/openhab/automation/jrule/internal/handler/JRuleEventHandler.java:253]: some more separation like "postUpdate '{}' on '{}'"

File: src/test/java/org/openhab/binding/jrule/internal/rules/timers/JRuleTimerTest.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        // Set item state in ItemRegistry
+        setState(new StringItem(JRuleTimerTestRules.TARGET_ITEM), UnDefType.UNDEF);
+        JRuleItemRegistry.get(JRuleTimerTestRules.TARGET_ITEM, TargetItem.class);
+        fireEvents(List.of(itemChangeEvent(JRuleTimerTestRules.TRIGGER_ITEM, "2", "1")));
+        verify(rule, times(1)).testSendCommand();
+        Thread.sleep(3000); // Wait for timer inside rule to execute
+        assertTrue(eventPublisher.hasCommandEvent(JRuleTimerTestRules.TARGET_ITEM, "command"));
+        assertTrue(eventPublisher.hasCommandEvent(JRuleTimerTestRules.TARGET_ITEM, "timedCommand"));
+        assertTrue(eventPublisher.hasCommandEvent(JRuleTimerTestRules.TARGET_ITEM, "nestedTimedCommand"));
+    }
```
Comment (querdenker2k) [inline src/test/java/org/openhab/binding/jrule/internal/rules/timers/JRuleTimerTest.java:50]: why or how does this work or test the timer execution?

File: src/test/java/org/openhab/binding/jrule/internal/rules/timers/TargetItem.java [hunk 5]
```diff
@@ -0,0 +1,42 @@
+/**
```
Comment (querdenker2k) [inline src/test/java/org/openhab/binding/jrule/internal/rules/timers/TargetItem.java:1]: Could be mocked or it is used multiple times?

**Review by querdenker2k — CHANGES_REQUESTED** [review]

File: src/test/java/org/openhab/binding/jrule/internal/rules/timers/JRuleTimerTest.java (hunk 4 above)
Comment (seime) [reply src/test/java/org/openhab/binding/jrule/internal/rules/timers/JRuleTimerTest.java:50]: The rule itself posts something to TARGET_ITEM when executed. It only tests the one-off timer execution.

Comment (seime) [reply src/test/java/org/openhab/binding/jrule/internal/rules/timers/JRuleTimerTest.java:50]: Could probably be mocked, just a bit messy to mock the static JRuleItemRegistry

File: src/test/java/org/openhab/binding/jrule/internal/rules/timers/TargetItem.java (hunk 5 above)
Comment (seime) [reply src/test/java/org/openhab/binding/jrule/internal/rules/timers/TargetItem.java:1]: Could probably be mocked, just a bit messy to mock the static JRuleItemRegistry

File: src/main/java/org/openhab/automation/jrule/internal/JRuleUtil.java (hunk 1 above)
Comment (seime) [reply src/main/java/org/openhab/automation/jrule/internal/JRuleUtil.java:99]: I think it is a good candidate for being static - as we only need one?

Comment (querdenker2k) [reply src/main/java/org/openhab/automation/jrule/internal/JRuleUtil.java:99]: all right, just asking. for thread-local it make sense to be static, but its used in multiple classes.
when you say its fine, its fine.

File: src/test/java/org/openhab/binding/jrule/internal/rules/timers/JRuleTimerTest.java (hunk 4 above)
Comment (querdenker2k) [reply src/test/java/org/openhab/binding/jrule/internal/rules/timers/JRuleTimerTest.java]: ahh, i thought its for timer triggers.

**Review by querdenker2k — APPROVED** [review]

**Comment by seaside1** [comment]
@seime  Do you want me to merge 74 or skip in favor of this?

**Comment by seime** [comment]
This one contains the commit from #74 so both can be merged I guess?

**Pull request merged**